check_codegen.sh

```diff
@@ -1,12 +1,29 @@
 # This is a script which checks if yarn codegen was run it used by the check_codegen evergreen task
-BEFORE=$(git status --porcelain | grep src/gql/generated/types.ts)
-yarn codegen
-AFTER=$(git status --porcelain | grep src/gql/generated/types.ts)
+
+# path to generated types
+types_file="src/gql/generated/types.ts"
+{
+  BEFORE=$(git status --porcelain | grep "$types_file")
+  yarn codegen
+  AFTER=$(git status --porcelain | grep "$types_file")
+} || {
+  echo "script failed"
+  exit 1
+}
```

Comment:
Should this be failing like it does on CI?

I changed this part of the script to this, which seemed to produce the correct results:

```suggestion
BEFORE=$(git status --porcelain | grep "$types_file")
yarn codegen || {
  echo "script failed"
  exit 1
}
AFTER=$(git status --porcelain | grep "$types_file")
```

[Passing Patch](https://spruce.mongodb.com/task/spruce_ubuntu1604_check_codegen_patch_16a6c01c5435829171c37bab8a45fb34f2dae505_62d856760ae60666474ddede_22_07_20_19_24_46/logs?execution=0)   |   [Failing Patch](https://spruce.mongodb.com/task/spruce_ubuntu1604_check_codegen_patch_16a6c01c5435829171c37bab8a45fb34f2dae505_62d857afc9ec4462ba4a6df9_22_07_20_19_29_56/logs?execution=0)

This [approach](https://github.com/minnakt/spruce/blob/alternate_codegen/check_codegen.sh) might also work 🤔 Not sure what's best